interactions/client.py

```diff
@@ -722,25 +763,117 @@ def __init__(self, client):
+                    func, name=func.__listener_name__
+                )  # capture the return value for friendlier ext-ing
+
+                listeners = self._listeners.get(func.__listener_name__, [])
+                listeners.append(func)
+                self._listeners[func.__listener_name__] = listeners
+
+            if hasattr(func, "__command_data__"):  # Set by extension_command
+                args, kwargs = func.__command_data__
+                func = client.command(*args, **kwargs)(func)
+
+                cmd_name = f"command_{kwargs.get('name') or func.__name__}"
+
+                commands = self._commands.get(cmd_name, [])
+                commands.append(func)
+                self._commands[cmd_name] = commands
+
+            if hasattr(func, "__component_data__"):
+                args, kwargs = func.__component_data__
+                func = client.component(*args, **kwargs)(func)
+
+                component = kwargs.get("component") or args[0]
+                comp_name = (
+                    _component(component).custom_id
+                    if isinstance(component, (Button, SelectMenu))
+                    else component
+                )
+                comp_name = f"component_{comp_name}"
+
+                components = self._components.get(comp_name, [])
+                components.append(func)
+                self._components[comp_name] = components
+
+        client.extensions[cls.__name__] = self
+
+        return self
+
+    def teardown(self):
+        for event, funcs in self._listeners.items():
+            for func in funcs:
+                self.client.websocket.dispatch.events[event].remove(func)
+
+        for component, funcs in self._components.items():
+            for func in funcs:
+                self.client.websocket.dispatch.events[component].remove(func)
+
+        for cmd, funcs in self._commands.items():
+            for func in funcs:
+                self.client.websocket.dispatch.events[cmd].remove(func)
+
+        clean_cmd_names = ["_".join(cmd.split("_")[1:]) for cmd in self._commands.keys()]
+        cmds = filter(
+            lambda x: x["name"] in clean_cmd_names,
+            self.client.http.cache.interactions.view,
+        )
+
+        if self.client.automate_sync:
+            [
+                self.client.loop.create_task(
+                    self.client.http.delete_application_command(
+                        cmd["application_id"], cmd["id"], cmd["guild_id"]
+                    )
+                )
+                for cmd in cmds
+            ]
+
+
+@functools.wraps(command)
+def extension_command(*args, **kwargs):
+    def decorator(coro):
+        coro.__command_data__ = (args, kwargs)
+        return coro
+
+    return decorator
+
+
+def extension_listener(name=None):
+    def decorator(func):
+        func.__listener_name__ = name or func.__name__
+
+        return func
+
+    return decorator
+
+
+# @functools.wraps(Client.component)
```

Comment:
```suggestion
@wraps(Client.component)
```

Please either uncomment this or remove it if you do not plan on using it.